app/services/resource_service.py

```diff
@@ -463,3 +473,18 @@ async def post_async_profile_analytics_request(request_body: object, params: Opt
     # TODO do error-handling over here
 
     return response
+
+
+@retry(attempts=3, delay=1, retry_exceptions=(TooManyRequestException,))
+async def post_async_contents_fetch_request(request_body: object, params: Optional[Dict]) -> Dict:
+
+    url = urllib.parse.urljoin(get_base_url(), "/v1/social/creators/async/contents/fetch")
+
+    response: Dict = await invoke_post_url(url=url,
```

Comment:
is it response object or response data, if it is response data then we should rename it to response_data

Reply:
This is response object.

Reply:
can you check again? I think it is response data, we seem to be extracting dict out of response object